var/spack/repos/builtin/packages/xsbench/package.py

```diff
@@ -18,29 +18,41 @@ class Xsbench(MakefilePackage):
 
     tags = ["proxy-app", "ecp-proxy-app"]
 
+    version("20", sha256="3430328267432b4c29605f248809caec3e8b0e3442d4dcd672fa09b8bb9aa1b6")
     version("19", sha256="57cc44ae3b0a50d33fab6dd48da13368720d2aa1b91cde47d51da78bf656b97e")
     version("18", sha256="a9a544eeacd1be8d687080d2df4eeb701c04eda31d3806e7c3ea1ff36c26f4b0")
     version("14", sha256="595afbcba8c1079067d5d17eedcb4ab0c1d115f83fd6f8c3de01d74b23015e2d")
     version("13", sha256="b503ea468d3720a0369304924477b758b3d128c8074776233fa5d567b7ffcaa2")
 
     variant("mpi", default=True, description="Build with MPI support")
     variant("openmp", default=True, description="Build with OpenMP support")
+    variant("cuda", default=False, when="@19:", description="Build with CUDA support")
 
     depends_on("mpi", when="+mpi")
 
+    conflicts("cuda_arch=none", when="+cuda", msg="Must select a CUDA architecture")
+    conflicts("+cuda", when="+openmp", msg="OpenMP must be disabled to support CUDA")
```

Comment:
Can we make the `openmp` variant conditional on `~cuda`? Would that make sense?

Reply:
I think that makes sense. XSBench is a mini-app so the goal would be to have all the different implementations installed and you can easily benchmark them. So all the variants are independent of each other and I just kept the default to OpenMP.